src/main/java/bio/terra/service/snapshot/flight/delete/DeleteSnapshotDeleteSamGroupStep.java

```diff
@@ -38,8 +42,8 @@ public StepResult doStep(FlightContext context) throws InterruptedException, Ret
         context
             .getWorkingMap()
             .get(SnapshotWorkingMapKeys.SNAPSHOT_AUTH_DOMAIN_GROUPS, new TypeReference<>() {});
-    // Only delete the Sam group if it matches the expected naming pattern
-    var expectedName = IamService.constructSamGroupName(snapshotId.toString());
+    // Only delete the Sam group if it matches the expected name
+    var expectedName = snapshotRequestDao.getByCreatedSnapshotId(snapshotId).samGroupName();
```

Comment:
I think this would throw an NPE on deleting non snapshot by request id snapshots.

Reply:
So true, thanks